src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/AbstractBolt.java

```diff
@@ -104,6 +109,18 @@ protected void dispatch(Tuple input) throws Exception {
 
     protected abstract void handleInput(Tuple input) throws Exception;
 
+    protected void handleLifeCycleEvent(LifecycleEvent event) {
+        if (Signal.START.equals(event.getSignal())) {
+            active = true;
+            emit(ZkStreams.ZK.toString(), currentTuple, new Values(event, commandContext));
```

Comment:
You are breaking stream responsibilities. In your implementation, the code emitting event in the stream do not relate with the code responsible for creating that stream with specific name and set of fields.

Delegate message emmitting into abstract method or define/create required stream(s) into this abstract bolt.

Reply:
I'll do a refactoring of all bolts in kilda in a separate pr, since they are not calling super.declareOutputFields() and declared this stream as a one of abstract bolts